### DISCUSSION TITLE
Documentation feedback: --cluster-init

### DISCUSSION THREAD

**Discussion by knweiss** [k3s-io/k3s#4395] [General]
The [Upgrades](https://rancher.com/docs/k3s/latest/en/upgrades/) section of the documentation currently reads _"To upgrade K3s from an older version you can re-run the installation script using the same flags"_.

However, when we set up a k3s "HA with embedded etcd db" configuration we still wondered if it is _really_ safe to specify `--cluster-init` for the **upgrade** runs, too. The worry was that this would reset ("initizalize") the etcd database.

**Comment by ChristianCiach** [comment]
It is not only safe, it is necessary. I've done it countless times. 

In fact, if you run k3s using a systemd unit (which is the case after installing k3s using the installation script), k3s gets started with `--cluster-init` every time anyway.

You can also use `--cluster-init` if your control plane (or even your whole cluster) only contains a single node. 

For these reasons I think that `--cluster-init` is a confusing name for this option. I think it should be named something like `--embedded-etcd`.

**Reply by brandond** [reply]
--cluster-init initializes a new etcd cluster if no etcd data is found on disk. The presence of an etcd datastore on disk supersedes all other options, ensuring that the cluster does not get reinitialized if already configured.

**Reply by gunzy83** [reply]
Just for my understanding: Does this mean that on upgrade the installer could just be run without the `--cluster-init` option and it would automatically pick up the etcd files on disk and rejoin the cluster?

**Reply by brandond** [reply]
Yep

**Reply by gunzy83** [reply]
Thanks!
